coderd/templates_test.go

```diff
@@ -145,6 +147,60 @@ func TestTemplateByOrganizationAndName(t *testing.T) {
 	})
 }
 
+func TestPatchTemplateMeta(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Modified", func(t *testing.T) {
+		t.Parallel()
+		ctx := context.Background()
+		client := coderdtest.New(t, nil)
+		user := coderdtest.CreateFirstUser(t, client)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID, func(ctr *codersdk.CreateTemplateRequest) {
+			ctr.Description = "original description"
+			ctr.MaxTTLMillis = ptr.Ref(24 * time.Hour.Milliseconds())
+			ctr.MinAutostartIntervalMillis = ptr.Ref(time.Hour.Milliseconds())
+		})
+		req := codersdk.UpdateTemplateMeta{
+			Description:                "lorem ipsum dolor sit amet et cetera",
+			MaxTTLMillis:               12 * time.Hour.Milliseconds(),
+			MinAutostartIntervalMillis: time.Minute.Milliseconds(),
+		}
+		updated, err := client.UpdateTemplateMeta(ctx, template.ID, req)
+		require.NoError(t, err)
+		assert.Greater(t, updated.UpdatedAt, template.UpdatedAt)
+		assert.Equal(t, req.Description, updated.Description)
+		assert.Equal(t, req.MaxTTLMillis, updated.MaxTTLMillis)
+		assert.Equal(t, req.MinAutostartIntervalMillis, updated.MinAutostartIntervalMillis)
```

Comment:
Suggestion (non-blocking): Should we additionally fetch the template to ensure that the propagation was written to the DB and not just propagated through `UpdateTemplateMeta`?

(This is a question of what-level-of-paranoid-do-we-want-to-be, I think it's fine as-is.)